index.js

```diff
@@ -59,15 +60,17 @@ class RNDraftView extends Component {
 
   widgetMounted = () => {
     const {
+      sourceType = 'html',
       placeholder,
       defaultValue,
       styleSheet,
       styleMap,
       blockRenderMap,
       onEditorReady = () => null
     } = this.props;
+    this.executeScript("setSourceType", sourceType);
     if (defaultValue) {
-      this.executeScript("setDefaultValue", defaultValue);
+      this.executeScript("setDefaultValue", defaultValue.replace(/(?:\r\n|\r|\n)/g, '\\n'));
```

Comment:
At this point, only the defaultValue gets escaped. But if you want a correct markdown output, on line 52, in the editorState every `\n` gets replaced by an empty string, resulting in 1 long line of markdown, instead of markdown with breaks.

This should only be applied if the `sourceType` is HTML.